packages/@cdktf/hcl2cdk/lib/expressions.ts

```diff
@@ -269,21 +300,36 @@ function convertTFExpressionAstToTs(
     }
 
     const rootSegment = segments[0].segment;
-    const subSegments =
-      rootSegment === "data" ? segments.slice(3) : segments.slice(2);
-    const indexOfNumericAccessor = subSegments.findIndex((seg) =>
+    const attributeIndex = rootSegment === "data" ? 3 : 2;
+    const attributeSegments = segments.slice(attributeIndex);
+    const indexOfNumericAccessor = attributeSegments.findIndex((seg) =>
```

Comment:
(optional) It appears like we could simplify this to be a boolean now like e.g. `hasNumericAccessor`. Or do we still use the index somewhere? A boolean would be simpler to read than `> -1` and `< 0` 🤔